sno/pk_generation.py

```diff
@@ -406,3 +452,45 @@ def feature_blobs(self):
             pk = self.decode_path_to_1pk(blob.name)
             if self.pk_filter(pk):
                 yield blob
+
+
+class FeaturePlusHashes:
+    _MORE_THAN_ONE_MATCH = object()
+
+    def __init__(self, feature, schema):
+        self.feature = feature
+        self.schema = schema
+        self.match = None
+
+    def __hash__(self):
+        return id(self)
+
+    def __eq__(self, other):
+        return other is self
+
+    def all_related_hashes(self):
+        value_hashes = []
+        feature_hash = 0
+        # Generate the overall hash of the feature, but in separate pieces.
+        for col in self.schema.non_pk_columns:
+            value = self.feature[col.name]
+            value_hash = hash(msg_pack([col.id, value]))
```

Comment:
does `msg_pack()` here add much value over just hashing `(col.id, value)`?

Reply:
Um... it's safe in that, all the values we store are guaranteed to message packable, and the hashes of message packed values are guaranteed to depend on their content, not their identity.
But, just looking through our types, the only weird one is geometry. And that's still just a glorified bytestring. So, you're right, we don't need to message pack here.

Reply:
Sure, feels like we could get another speed up avoiding the encoding.

Even for a `Geometry` type we could add `__hash__()` that returns `hash(self.wkb)` or similar.